common/primitives/src/messages.rs

```diff
@@ -13,15 +13,18 @@ use utils::*;
 /// A type for responding with an single Message in an RPC-call.
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 #[derive(Default, Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq)]
-pub struct MessageResponse<AccountId, BlockNumber> {
+pub struct MessageResponse<BlockNumber> {
 	#[cfg_attr(feature = "std", serde(with = "as_hex"))]
 	/// Serialized data in a the schemas.
 	pub payload: Vec<u8>,
-	/// The public key of the provider and the signer of the transaction.
-	pub provider_key: AccountId,
-	/// Message source account id (the original source).
-	pub msa_id: MessageSourceId,
-	/// Index in block to get total order
+	/// Message source account id of the MSA that signed the transaction. This may be the same id
+	/// as contained in `original_msa_id`, indicating that the transaction to add the message was
+	/// signed by the originator. An id differing from that of `original_msa_id` indicates that a
+	/// provider was delegated by `original_msa_id` to send this message on its behalf.
+	pub signer_msa_id: MessageSourceId,
+	///  Message source account id (the original source).
+	pub original_msa_id: MessageSourceId,
```

Comment:
Hmm.... I'm not sure about this naming over:
- `source_msa_id` or just `msa_id`
- `provider_msa_id`

Mostly to match the official data model:
<img width="370" alt="image" src="https://user-images.githubusercontent.com/1252199/183417879-a7d76e3f-5875-450f-bac0-5dfeefb9393c.png">

https://github.com/LibertyDSNP/frequency/tree/main/designdocs

Reply:
Hmm... also what about `origin` instead of `original` or `source`?

Reply:
`provider_msa_id` didn't make sense to me to use unless there will always be a delegator / provider relationship. I suggested `signer_msa_id` because it signifies the holder of the MSA that signed the transaction to add the message. If however, we document very well that an MSA can be its own provider in the case that there is no delegation to a different MSA, then `provider_msa_id` makes sense to me (I am thinking of confusion for new developers).

`msa_id` instead of `original_msa_id` works for me also, `original` doesn't seem to add much value to the field name.

Reply:
I'm not sure it is any more or less documentation than what we already have to have around signer_msa_id. That introduces more questions. Who can be a signer? What if the signer is someone that isn't delegated to by the msa_id?

provider_msa_id deals with a lot of those issues. And the documentation around being one's own provider I think would be minimal (especially as it should be uncommon).

Reply:
Good points; I will refactor the variable names as:
`signer_msa_id`  -> `provider_msa_id`
`original_msa_id` -> `msa_id`

Also will update design docs to signify that instead of a provider key, we will store the provider msa on the `Message`